code/game/turfs/turf.dm

```diff
@@ -554,7 +557,7 @@
 				continue
 			if(O.invisibility == 101)
 				O.singularity_act()
-	ChangeTurf(/turf/space)
+	ChangeTurf(get_base_turf(src.z))
```

Comment:
same, it'll keep eating it for infinity

@dylanstrategie, you could make the singuloth not eat the base turf